src/Common/src/System/IO/PathInternal.Unix.cs

```diff
@@ -23,6 +24,11 @@ internal static int GetRootLength(string path)
             return path.Length > 0 && IsDirectorySeparator(path[0]) ? 1 : 0;
         }
 
+        internal static int GetRootLength(ReadOnlySpan<char> path)
+        {
+            return path.Length > 0 && IsDirectorySeparator(path[0]) ? 1 : 0;
+        }
```

Comment:
I can't comment directly on it, but below there's a use of StringBuilder.  Subsequently, we should look at such usage and see whether it makes sense to switch over to the ValueStringBuilder I recently added.  It's a struct, can start out using stack space, and uses arrays from ArrayPool.Default.

Reply:
Good to know- I'll look into it.